src/vs/editor/contrib/hover/browser/contentHover.ts

```diff
@@ -341,16 +359,21 @@ export class ContentHoverController extends Disposable {
 			}
 		}
 
+		const options = { mode: HoverStartMode.Delayed, focus: false };
 		if (anchorCandidates.length === 0) {
-			return this._startShowingOrUpdateHover(null, HoverStartMode.Delayed, HoverStartSource.Mouse, false, mouseEvent);
+			return this._startShowingOrUpdateHover(null, HoverStartSource.Mouse, mouseEvent, options);
 		}
 
 		anchorCandidates.sort((a, b) => b.priority - a.priority);
-		return this._startShowingOrUpdateHover(anchorCandidates[0], HoverStartMode.Delayed, HoverStartSource.Mouse, false, mouseEvent);
+		return this._startShowingOrUpdateHover(anchorCandidates[0], HoverStartSource.Mouse, mouseEvent, options);
+	}
+
+	public startShowingAtRange(range: Range, source: HoverStartSource, opts: IHoverOptions): void {
```

Comment:
This seems to be a good refactoring, but it would be nice if the refactoring was seperate from the feature, so that reviewing the change is easier.
When your PR is ready, I can recommend to rewrite your commit history for that!